manifest: order subsystems alphabetically

benediktibk opened the pull request:
Order the subsystems in the manifest file alphabetically, as they actually should be according to the comment within the manifest file.

Signed-off-by: Benedikt Schmidt <benedikt.schmidt@embedded-solutions.at>

erwango commented on west.yml:
```diff
@@ -25,46 +25,59 @@ manifest:
     - name: hal_infineon
       revision: f1fa8241f8786198ba41155413243de36ed878a5
       path: modules/hal/infineon
+    - name: lvgl
+      revision: 31acbaa36e9e74ab88ac81e3d21e7f1d00a71136
+      path: modules/lib/gui/lvgl
```
lvgl should not be there

benediktibk commented:
Sorry for the multiple updates, I'm currently trying to get to know this ecosystem and learning how I can contribute something.

galak commented:
> Sorry for the multiple updates, I'm currently trying to get to know this ecosystem and learning how I can contribute something.

No worries, thanks for contributing!

mbolivar-nordic reviewed (commented):
This needs a rebase due to the following change:

```

-hal_nxp      modules/hal/nxp              73445ce7474ffb0eb6fd795eeaac1fbf507e4b7e https://github.com/zephyrproject-rtos/hal_nxp
+hal_nxp      modules/hal/nxp              0d11138724959e1162777d9206f841ccdf64348e https://github.com/zephyrproject-rtos/hal_nxp
```

benediktibk committed:
manifest: order subsystems alphabetically

Order the subsystems in the manifest file alphabetically, as they
actually should be according to the comment within the manifest file.

Signed-off-by: Benedikt Schmidt <benedikt.schmidt@embedded-solutions.at>

benediktibk commented:
Do I have to trigger the checks "Run compliance checks on patch series (PR)" and "Scan code for licenses" manually?

galak commented:
> Do I have to trigger the checks "Run compliance checks on patch series (PR)" and "Scan code for licenses" manually?

I triggered that for you.  Github "feature" that for first time contributors we have to click a button to do it.
